core/src/main/java/org/apache/iceberg/MergingSnapshotUpdate.java

```diff
@@ -46,19 +47,17 @@
 import org.apache.iceberg.util.CharSequenceWrapper;
 import org.apache.iceberg.util.StructLikeWrapper;
 import org.apache.iceberg.util.Tasks;
+import org.apache.iceberg.util.ThreadPools;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import static com.google.common.collect.Iterables.filter;
-import static com.google.common.collect.Iterables.transform;
 import static org.apache.iceberg.TableProperties.MANIFEST_MIN_MERGE_COUNT;
 import static org.apache.iceberg.TableProperties.MANIFEST_MIN_MERGE_COUNT_DEFAULT;
 import static org.apache.iceberg.TableProperties.MANIFEST_TARGET_SIZE_BYTES;
 import static org.apache.iceberg.TableProperties.MANIFEST_TARGET_SIZE_BYTES_DEFAULT;
-import static org.apache.iceberg.util.ThreadPools.getWorkerPool;
 
 abstract class MergingSnapshotUpdate extends SnapshotUpdate {
-  private final Logger LOG = LoggerFactory.getLogger(getClass());
+  private static final Logger LOG = LoggerFactory.getLogger(MergingSnapshotUpdate.class);
```

Comment:
I think the idea was to log from the concrete class, not from `MergingSnapshotUpdate` so that it is clear what operation was used from the logs. It isn't a big deal, but I think that it makes some sense to do that. Other opinions?

Reply:
I'm strongly against passing `Logger` instances around since it leads to a lot of difficult to trace bugs. Particularly, it's possible for the logger to be tied to this class (since we call `getClass`) but then it gets passed to a different object of a different class, thus making the logger context inconsistent.

Reply:
I don't think we are passing a logger around. It is private and only available here. But instead of logging from MergingSnapshotUpdate, logs come from MergeAppend or RewriteFiles. I thought that was more clear, but I'm not strongly against changing this. Let's keep this change.